core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3190,6 +3140,95 @@ private boolean syncMetadata(JournalContext journalContext, LockedInodePath inod
     return true;
   }
 
+  private boolean syncDirMetadata(JournalContext journalContext, LockedInodePath inodePath,
+      int syncDescendantLevels)
+      throws FileDoesNotExistException, InvalidPathException {
+    if (syncDescendantLevels == 0) {
+      return false;
+    }
+    MountTable.Resolution resolution = mMountTable.resolve(inodePath.getUri());
+    AlluxioURI ufsUri = resolution.getUri();
+    UnderFileSystem ufs = resolution.getUfs();
+
+    boolean loadMetadata = false;
+    DeleteOptions syncDeleteOptions =
+        DeleteOptions.defaults().setRecursive(true).setAlluxioOnly(true).setUnchecked(true);
+    Inode<?> inode = inodePath.getInode();
+
+    UfsStatus[] listStatus = null;
+    try {
+      listStatus = ufs.listStatus(ufsUri.toString());
+    } catch (IOException e) {
+      // ignore error
+      LOG.warn("Failed to list directory: {} error: {}", ufsUri, e.toString());
```

Comment:
Does this message need more detail, since we list in many places and why it is a harmless error?